### PR TITLE
Fixing ScrubberEditor segfault on wx

### LINKED ISSUE #434: ScrubberEditor segfault on Wx
Get segfault on initial mouseover when running `Scrubber_editor_demo.py`

### LINKED ISSUE #1066: Problem with Scrubber_editor_demo.py
Running on wx gives 

<details>

```
$ python Scrubber_editor_demo.py 
01:03:59 PM: Debug: Adding duplicate image handler for 'Windows bitmap file'
01:03:59 PM: Debug: Adding duplicate animation handler for '1' type
01:03:59 PM: Debug: Adding duplicate animation handler for '2' type
01:03:59 PM: Debug: Adding duplicate image handler for 'Windows bitmap file'
01:03:59 PM: Debug: Adding duplicate animation handler for '1' type
01:03:59 PM: Debug: Adding duplicate animation handler for '2' type
01:03:59 PM: Debug: Adding duplicate image handler for 'Windows bitmap file'
01:03:59 PM: Debug: Adding duplicate animation handler for '1' type
01:03:59 PM: Debug: Adding duplicate animation handler for '2' type
Traceback (most recent call last):
  File "/Users/aayres/.edm/envs/wx-env/lib/python3.6/site-packages/traitsui/wx/view_application.py", line 152, in OnInit
    args=self.args,
  File "/Users/aayres/.edm/envs/wx-env/lib/python3.6/site-packages/traitsui/view.py", line 462, in ui
    ui.ui(parent, kind)
  File "/Users/aayres/.edm/envs/wx-env/lib/python3.6/site-packages/traitsui/ui.py", line 246, in ui
    self.rebuild(self, parent)
  File "/Users/aayres/.edm/envs/wx-env/lib/python3.6/site-packages/traitsui/wx/toolkit.py", line 125, in ui_live
    ui_live.ui_live(ui, parent)
  File "/Users/aayres/.edm/envs/wx-env/lib/python3.6/site-packages/traitsui/wx/ui_live.py", line 52, in ui_live
    _ui_dialog(ui, parent, BaseDialog.NONMODAL)
  File "/Users/aayres/.edm/envs/wx-env/lib/python3.6/site-packages/traitsui/wx/ui_live.py", line 88, in _ui_dialog
    BaseDialog.display_ui(ui, parent, style)
  File "/Users/aayres/.edm/envs/wx-env/lib/python3.6/site-packages/traitsui/wx/ui_base.py", line 71, in display_ui
    ui.owner.init(ui, parent, style)
  File "/Users/aayres/.edm/envs/wx-env/lib/python3.6/site-packages/traitsui/wx/ui_live.py", line 203, in init
    sw = panel(ui, window)
  File "/Users/aayres/.edm/envs/wx-env/lib/python3.6/site-packages/traitsui/wx/ui_panel.py", line 272, in panel
    panel, content[0], ui
  File "/Users/aayres/.edm/envs/wx-env/lib/python3.6/site-packages/traitsui/wx/ui_panel.py", line 431, in fill_panel_for_group
    panel, group, ui, suppress_label, is_dock_window, create_panel
  File "/Users/aayres/.edm/envs/wx-env/lib/python3.6/site-packages/traitsui/wx/ui_panel.py", line 585, in __init__
    self.add_groups(content, panel)
  File "/Users/aayres/.edm/envs/wx-env/lib/python3.6/site-packages/traitsui/wx/ui_panel.py", line 729, in add_groups
    sizer.Add(sg_sizer, growable, style, 2)
wx._core.wxAssertionError: C++ assertion "!(flags & (wxALIGN_BOTTOM | wxALIGN_CENTRE_VERTICAL))" failed at /Users/robind/projects/bb2/dist-osx-py36/build/ext/wxWidgets/src/common/sizer.cpp(2114) in DoInsert(): Vertical alignment flags are ignored with wxEXPAND
OnInit returned false, exiting...
```

</details>

and on qt:
<details>

```
$ python Scrubber_editor_demo.py 
Traceback (most recent call last):
  File "Scrubber_editor_demo.py", line 165, in <module>
    demo.configure_traits()
  File "/Users/aayres/.edm/envs/qt-env/lib/python3.6/site-packages/traits/has_traits.py", line 2015, in configure_traits
    args,
  File "/Users/aayres/.edm/envs/qt-env/lib/python3.6/site-packages/traitsui/qt4/toolkit.py", line 232, in view_application
    context, view, kind, handler, id, scrollable, args
  File "/Users/aayres/.edm/envs/qt-env/lib/python3.6/site-packages/traitsui/qt4/view_application.py", line 84, in view_application
    context, view, kind, handler, id, scrollable, args
  File "/Users/aayres/.edm/envs/qt-env/lib/python3.6/site-packages/traitsui/qt4/view_application.py", line 126, in __init__
    args=self.args,
  File "/Users/aayres/.edm/envs/qt-env/lib/python3.6/site-packages/traitsui/view.py", line 462, in ui
    ui.ui(parent, kind)
  File "/Users/aayres/.edm/envs/qt-env/lib/python3.6/site-packages/traitsui/ui.py", line 246, in ui
    self.rebuild(self, parent)
  File "/Users/aayres/.edm/envs/qt-env/lib/python3.6/site-packages/traitsui/qt4/toolkit.py", line 162, in ui_live
    ui_live.ui_live(ui, parent)
  File "/Users/aayres/.edm/envs/qt-env/lib/python3.6/site-packages/traitsui/qt4/ui_live.py", line 43, in ui_live
    _ui_dialog(ui, parent, BaseDialog.NONMODAL)
  File "/Users/aayres/.edm/envs/qt-env/lib/python3.6/site-packages/traitsui/qt4/ui_live.py", line 65, in _ui_dialog
    BaseDialog.display_ui(ui, parent, style)
  File "/Users/aayres/.edm/envs/qt-env/lib/python3.6/site-packages/traitsui/qt4/ui_base.py", line 284, in display_ui
    ui.owner.init(ui, parent, style)
  File "/Users/aayres/.edm/envs/qt-env/lib/python3.6/site-packages/traitsui/qt4/ui_live.py", line 225, in init
    self.add_contents(panel(ui), bbox)
  File "/Users/aayres/.edm/envs/qt-env/lib/python3.6/site-packages/traitsui/qt4/ui_panel.py", line 259, in panel
    panel = _GroupPanel(content[0], ui).control
  File "/Users/aayres/.edm/envs/qt-env/lib/python3.6/site-packages/traitsui/qt4/ui_panel.py", line 599, in __init__
    layout = self._add_groups(content, inner)
  File "/Users/aayres/.edm/envs/qt-env/lib/python3.6/site-packages/traitsui/qt4/ui_panel.py", line 680, in _add_groups
    panel = _GroupPanel(subgroup, self.ui).control
  File "/Users/aayres/.edm/envs/qt-env/lib/python3.6/site-packages/traitsui/qt4/ui_panel.py", line 601, in __init__
    layout = self._add_items(content, inner)
  File "/Users/aayres/.edm/envs/qt-env/lib/python3.6/site-packages/traitsui/qt4/ui_panel.py", line 867, in _add_items
    ui, object, name, item.tooltip, None
  File "/Users/aayres/.edm/envs/qt-env/lib/python3.6/site-packages/traitsui/editor_factory.py", line 137, in simple_editor
    description=description,
  File "/Users/aayres/.edm/envs/qt-env/lib/python3.6/site-packages/pyface/base_toolkit.py", line 184, in __init__
    raise NotImplementedError(msg % (toolkit, package, name))
NotImplementedError: the qt4 traitsui.qt4 backend doesn't implement scrubber_editor:_ScrubberEditor
```

</details>

### PR DESCRIPTION
**Pull request by aaronayres35**
fixes #1066 
fixes #434

### CODE REVIEW

**Comment by aaronayres35** [comment]
as described in the issue #1066 the error seems to be from calling setFocus() multiple times in a row.  I tried adding a check in the _enter_window method to prevent calling `self.control.SetFocus()` if it has already been called for text.  This seems to fix the issue - I no longer see seagulls when I click a text box exit, and then mouseover the box again like before.  However, after clicking and moving around the textboxes I no see different set faults on occasion.  I currently am unable to reproduce when they occur - trying to figure that out.

**Comment by aaronayres35** [comment]
In the latest commit, there is no longer a seagull after clicking a text box, leaving and mousing over any other text box.  However, if you click to type in one, leave and the mouseover and *click* on another, it segfaults.

**Comment by aaronayres35** [comment]
After the last fix I can't make it segfault!  Unfortunately its still broken, as now once you click on a textbox it can only be used for text.  You can't click and drag again to use it as a scrubber, unless you recheck into and hit enter.

File: traitsui/wx/scrubber_editor.py [hunk 1]
```diff
@@ -346,6 +346,7 @@ def _pop_up_text(self):
         text.SetFocus()
         control.Bind(wx.EVT_TEXT_ENTER, self._text_completed, id=text.GetId())
         text.Bind(wx.EVT_KILL_FOCUS, self._text_completed)
+        text.Bind(wx.EVT_KILL_FOCUS, self._left_down)
```
Comment (kitchoi) [inline traitsui/wx/scrubber_editor.py:349]: Thank you for all these efforts tracking down and fixing the segfault.

Could you please explain how this works (useful for future references too)?
For instance, the event emitted by `EVT_KILL_FOCUS` is an instance of [`wx.FocusEvent`](https://wxpython.org/Phoenix/docs/html/wx.FocusEvent.html#wx-focusevent).
I don't think the event supports the `GetX()` and `GetY()` methods needed by `self._left_down`, so I am quite confused...

Comment (kitchoi) [reply traitsui/wx/scrubber_editor.py:349]: I think I might know why this seems to fix the issue...
It _seems_ the wx calls event handlers in the reverse order of when they are bound (I don't know if the order is documented or not).

For example, if I swap the order of `text.Bind(wx.EVT_KILL_FOCUS, self._text_completed)` and `text.Bind(wx.EVT_KILL_FOCUS, self._left_down)`, and I will get the segfault.

Here, the fact that `_left_down` causes an exception with `FocusEvent` prevented `self._text_completed` from being called, and the segfault seems to come from `self._text_completed`.

Comment (kitchoi) [reply traitsui/wx/scrubber_editor.py:349]: With that understanding... adding `return` in `self._text_completed` after `event.Skip()` seems to achieve the same effect as the changes added here.
IOW, this:
```
    def _text_completed(self, event):
        """ Handles the user pressing the 'Enter' key in the text control.
        """
        if isinstance(event, wx.FocusEvent):
            event.Skip()
            return        # <---- here
        if self._update_value(event):
            self._destroy_text()
```

Comment (kitchoi) [reply traitsui/wx/scrubber_editor.py:349]: In fact... that would be the same as removing `text.Bind(wx.EVT_KILL_FOCUS, self._text_completed)` altogether :)

While this results in a reduction in feature (the editor is probably meant to replicate the effect of hitting enter on the text box while it loses focus), the feature does not work and causes segfault. By removing it, we at least resolve the crippling segfault. The removed feature can be added back separately if we ever need it back.

File: traitsui/wx/scrubber_editor.py [hunk 2]
```diff
@@ -461,13 +459,12 @@ def _enter_window(self, event):
...
         self.control.SetCursor(wx.Cursor(wx.CURSOR_HAND))
-
+        
         if not self._ignore_focus:
             self._ignore_focus = True
             self.control.SetFocus()
-
         self._ignore_focus = False
-
+
```
Comment (kitchoi) [inline traitsui/wx/scrubber_editor.py:467]: Not a request change, just a comment for future consideration:
Flake8 will complain about these trailing whitespaces (yep we should run flake8 in CI, but we don't at the moment). I think my IDE actually automatically removes trailing whitespaces for me when I save a file, might be something that is useful to set up once and for all.

**Pull request merged**